src/documents/tutorial-developerIntroduction/DeveloperIntroductionToInfusionFramework-OpeningRemarks.md

```diff
@@ -0,0 +1,598 @@
+ * [`fluid.modelComponent`](/infusion/development/ComponentConfigurationOptions.md#model-components), which adds support for mutable state, model events, and state coordination and relay between different components.
+ * [`fluid.viewComponent`](/infusion/development/ComponentConfigurationOptions.md#view-components), which supplement model components with functionality to bind to the DOM of a web page.
+
+Component-based development emphasizes declarative configuration, loose coupling and flexible hierarchies. The framework manages component lifecycle and scoping and provides many supports to allow components to work in collaboration with one another.
+
+A new Infusion component is defined using the `fluid.defaults` function and a very basic "Hello, World!" component might look like this:
+
+<div class="infusion-docs-note">You can check out the [Live Example](http://codepen.io/waharnum/pen/oBYvwx?editors=1111) of the code below on [CodePen](http://codepen.io/)</div>
+
+
+```
+// The first argument is the name of the new component, the second is the
+// new component's default configuration
+fluid.defaults("fluid.helloWorld", {
+    // gradeNames supplies an array of strings that name previously
+    // defined Infusion components
+    gradeNames: ["fluid.component"]
+});
+```
+
+After definition, instances of the component can be created by calling the component's name as a function, with the option to pass in further configuration options as an argument:
+
+```
+var helloWorld = fluid.helloWorld();
+```
+
+Right now this component doesn't do anything new, but we will evolve its definition throughout this introduction to demonstrate further core concepts of the framework.
+
+### Invokers
+
+All Infusion components support the definition of public functions using [invokers](/infusion/development/Invokers.md); among other things, invokers allow related components to have different internal implementations of a function but present a common function to be called by other code. If we imagine a "Hello, World!" component, there are a variety of ways it might say "hello". Two of them are:
+
+* By printing to the web developer console via `console.log`
+* By displaying a message on a web page
+
+We'll define the first approach as an invoker on the "Hello, World!" component. Other styles of invokers are possible, but we'll use one here that lets us refer to functions of existing Javascript objects:
+
+<div class="infusion-docs-note">You can check out the [Live Example](http://codepen.io/waharnum/pen/MJbgEx?editors=1111) of the code below on [CodePen](http://codepen.io/)</div>
+
+```
+fluid.defaults("fluid.helloWorld", {
+    gradeNames: ["fluid.component"],
+    invokers: {
+        sayHello: {
+            // Configures this invoker to use the console object
+            "this": "console",
+            // Configures this invoker to use the log function of the console
```

Comment:
Pretending to be a novice JavaScript programmer here and asking intentionally naive questions:  The use of `"this"` is puzzling.  What is `"this"` in this context?  Why is it used?

What first pops into mind is that `"this"` refers to an instance of a `helloWorld` object created at run-time.  But that can't be right since `helloWorld` doesn't have a `console` member.  As it happens, in this case, `console` is a member of the `window` object in the browser, so the next question is: why not use `window` instead of `"this"`?

I've tried to come up with a concise transparent explanation of what `"this"` means here, but I'm stumped.

A related overall comment:
It would be useful to note explicitly that `gradenames`, `invokers`, `listeners`, etc. are reserved words in infusion, although I'm not sure that "reserved word" is the correct term here.  Perhaps, "behave like reserved words"?  Anyhow, these are fixed and have special meaning within both infusion and the `fluid.defaults()` function.  Similarly, within an  invoker block, `"this"`, `"method"`, `"args"`, "`func`", and "`funcName"` are reserved.  There are others like `"model"`, etc.

It would be useful to call out when the left side of the colon is defined by the framework (e.g., `invokers`), and when it's up the programmer (e.g., `sayHello`).